File: examples/motion/linear.cpp
#include "pyro/pyro.h"
#include <iostream>

using namespace Pyro;

float a;

void kb_keypressed()
{
    std::cout << " A key was pressed...." << std::endl;
}

void setup()
{
    size(640, 360);
    stroke(255);
    a = height / 2;

    set_keypressed(&kb_keypressed);
}

void draw()
{
    background(51);
    line(0, a, width, a);
    a = a - 0.5;
    if (a < 0)
    {
        a = height;
    }
}

int main(int argc, char **argv)
{
    run(setup, draw);
    return 0;
}

File: examples/motion/linear.py
import pyro

a = 0

def setup():
    pyro.size(640, 360)
    pyro.stroke(1.0)

def draw():
    global a
    pyro.background(51 / 255.0)
    pyro.line(0, a, pyro.width, a)
    a = a - 0.5
    if a < 0:
        a = pyro.height

pyro.run(setup, draw)
